contracts/MigrationContractStaging.cdc

```diff
@@ -0,0 +1,362 @@
+/// This contract is intended for use in the Cadence 1.0 contract migration across the Flow network.
+///
+/// In preparation for this milestone, your contract will NEED to be updated! Once you've updated your code for
+/// Cadence 1.0, you MUST stage your contract code in this contract so that the update can be executed as a part of the
+/// network-wide state migration.
+///
+/// To stage your contract update:
+/// 1. create a Host & save in your contract-hosting account
+/// 2. call stageContract() passing a reference to you Host, the contract name, and the hex-encoded Cadence code
+///
+/// This can be done in a single transaction! For more code context, see https://github.com/onflow/contract-updater
+///
+access(all) contract MigrationContractStaging {
+
+    // Path constants
+    //
+    access(self) let delimiter: String
+    access(self) let capsulePathPrefix: String
+    access(all) let HostStoragePath: StoragePath
+    access(all) let AdminStoragePath: StoragePath
+    /// Maps contract addresses to an array of staged contract names
+    access(self) let stagedContracts: {Address: [String]}
+    /// The block height at which updates can no no longer be staged. If nil, updates can be staged indefinitely until
+    /// the cutoff value is set.
+    access(all) var stagingCutoff: UInt64?
+
+    /// Event emitted when a contract's code is staged
+    /// `action` ∈ {"stage", "replace", "unstage"} each denoting the action being taken on the staged contract
+    /// NOTE: Does not guarantee that the contract code is valid Cadence
+    access(all) event StagingStatusUpdated(
+        capsuleUUID: UInt64,
+        address: Address,
+        codeHash: [UInt8],
+        contract: String,
+        action: String
```

Comment:
Given the description in the comment, maybe just define an enum for it to make invalid states unrepresentable

Reply:
I wondered that, but I don't think I can emit an event can I? If not, I could emit its raw value, but I wonder if that would be preferable for systems monitoring these events - @bjartek @sideninja any thoughts?

Reply:
I confirmed I'm not able to emit an enum, it would just be it's raw value - 0, 1, 2

Reply:
Then i prefer a string